test/ecto/adapters/myxql_test.exs

```diff
@@ -281,6 +281,21 @@ defmodule Ecto.Adapters.MyXQLTest do
       ~s{SELECT GROUP_CONCAT(st0.`id` SEPARATOR ' / ') AS `breadcrumbs` FROM `tree` AS st0) AS s1 ON TRUE}
   end
 
+  test "parent binding subquery and combination" do
+    right_query = from(c in "categories_right", where: c.id == parent_as(:c).id, select: c.id)
+    left_query = from(c in "categories_left", where: c.id == parent_as(:c).id, select: c.id)
+    union_query = union(left_query, ^right_query)
+    query = from(c in "categories", as: :c, where: c.id in subquery(union_query), select: c.id) |> plan()
+
+    assert all(query) ==
+      ~s{SELECT c0.`id` FROM `categories` AS c0 } <>
+      ~s{WHERE (} <>
+      ~s{c0.`id` IN } <>
+      ~s{(SELECT sc0.`id` FROM `categories_left` AS sc0 WHERE (sc0.`id` = c0.`id`) } <>
+      ~s{UNION } <>
+      ~s{(SELECT c0.`id` FROM `categories_right` AS c0 WHERE (c0.`id` = c0.`id`))))}
```

Comment:
Actually there is another issue here. The aliases are conflicting probably because it's not considering the combination query as part of the subquery. I will look into this.

Reply:
I fixed this by passing `as_prefix` into the combination function. I believe we don't need to add an extra `c` because the only chance for a name conflict is if it's referencing a parent query from inside of a subquery.